### PR TITLE
Update AzureML model deploy 

### PR DESCRIPTION
**Pull request by shivp950**
Signed-off-by: Shivani Patel 41974238+shivp950@users.noreply.github.com

## What changes are proposed in this pull request?

Removed the build image method for deploying an MLflow model to AzureML since it is deprecated. I added the new azureml.mlflow.deploy method.

## How is this patch tested?
Azure ML team developed and validated this method with internal tests

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

Model Deployments to Azure Machine Learning do not require an image build step anymore, there's a new azureml.mlflow.deploy method. With the new method, users can package and deploy a model in one step. 

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [x] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [x] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by shivp950** [comment]
@akshaya-a since I was blocked, I just closed the other PR and added all the changes to this one so we can get this merged quicker

**Comment by jpe316** [comment]
Package and deploy a model*

**Review by jpe316 — APPROVED** [review]

**Comment by shivp950** [comment]
@dbczumar I was unable to work around the DCO sign off issue since the force push removed the latest commit. I've added all the feedback from this PR(https://github.com/mlflow/mlflow/pull/2899#issue-427024464), could you please take a look? Thanks again for all your help!

File: docs/source/models.rst [hunk 1]
```diff
@@ -829,23 +822,26 @@ accepts the following data formats as input:
...
     # webservice's scoring URI. The following example posts a sample input from the wine dataset
     # used in the MLflow ElasticNet example:
     # https://github.com/mlflow/mlflow/tree/master/examples/sklearn_elasticnet_wine
-
+    
     scoring_uri=$(az ml service show --name <deployment-name> -v | jq -r ".scoringUri")
 
-    # `sample_input` is a JSON-serialized pandas DataFrame with the `split` orientation
+    
+        # `sample_input` is a JSON-serialized pandas DataFrame with the `split` orientation
```
Comment (dbczumar) [inline docs/source/models.rst:844]: Looks like this is a bit over indented - can we align it with the code below?

File: docs/source/models.rst [hunk 2]
```diff
@@ -781,30 +780,24 @@ accepts the following data formats as input:
...
+    # After the model deployment completes, requests can be posted via HTTP to the new ACI
     # webservice's scoring URI. The following example posts a sample input from the wine dataset
     # used in the MLflow ElasticNet example:
     # https://github.com/mlflow/mlflow/tree/master/examples/sklearn_elasticnet_wine
     print("Scoring URI is: %s", webservice.scoring_uri)
 
     import requests
     import json
-
+
```
Comment (dbczumar) [inline docs/source/models.rst:800]: Can we revert these additions of empty whitespace? Looks like there are several lines where we added unneeded whitespace.

**Review by dbczumar — COMMENTED** [review]
@shivp950 Looks great! Left a couple of nit comments. Will LGTM and merge after they are addressed.

File: docs/source/models.rst (hunk 2 above)
Comment (shivp950) [reply docs/source/models.rst:800]: will do

File: docs/source/models.rst (hunk 1 above)
Comment (shivp950) [reply docs/source/models.rst:844]: definitely

File: docs/source/models.rst (hunk 2 above)
Comment (dbczumar) [reply docs/source/models.rst:801]: @shivp950 Looks like these instances of empty whitespace are still present. Can we remove them?

Comment (shivp950) [reply docs/source/models.rst:801]: I did it in the 3rd commit then reverted it since I ran into a failure on the examples/examples test. I'll fix the space again and hopefully the error goes away

File: docs/source/models.rst [hunk 3]
```diff
@@ -829,22 +822,23 @@ accepts the following data formats as input:
                   headers={"Content-type": "application/json"})
     response_json = json.loads(response.text)
     print(response_json)
-
+
```
Comment (dbczumar) [inline docs/source/models.rst:825]: Looks like we still have unnecessary whitespace. More than happy to push a commit to remove this - let me know!

Comment (shivp950) [reply docs/source/models.rst]: thanks for pushing the commit! I appreciate it, however there's still a test failing, let me know what I can do. I don't fully understand the error

Comment (harupy) [reply docs/source/models.rst]: @shivp950 It looks like the r test is a bit flaky. I'll rerun the CI checks.

Comment (shivp950) [reply docs/source/models.rst]: @harupy thank you!

Comment (shivp950) [reply docs/source/models.rst]: @dbczumar and @harupy if this pr looks good to you, can you please approve it when you get the chance?

**Review by dbczumar — APPROVED** [review]
LGTM!
